doc/vxlan/data_plane_VXLAN_HLD.md

```diff
@@ -0,0 +1,507 @@
+
+##### 5.2 Remote VTEP-VNI association configuration 
+
+* Ability to add and remove remote VTEP assiciation to VNI
+   - config vxlan remote map add <vtepname> <vni-id>
+   - config vxlan remote map del <vtepname> <vni-id>
+   - vtepname is a string. 
+   - ipv4-address is an IPV4 address in dotted notation A.B.C.D
+
+#### Yang Enhancements
+
+##### VXLAN_REMOTE_VTEP
+
+container VXLAN_REMOTE_VTEP {
+
+    description "config db VXLAN_REMOTE_VTEP table";
+
+    list VXLAN_REMOTE_VTEP_LIST {
+
+        key "name";
+        max-elements 1;
+
+        leaf name {
+        /* vni devices are created of the form 'name'-vlanid
+            The kernel has a max limit of 15 chars for netdevices.
+            keeping aside 5 chars for hyphen and vlanid the
+            name should have a max of 10 chars */
+
+            type string {
+                length 1..10;
+            }
+        }
+
+        leaf src_ip {
+            type inet:ip-address;
+        }
+    }
+}
+
+##### VXLAN_REMOTE_VTEP_VNI
+
+container VXLAN_REMOTE_VTEP_VNI {
+
+    description "config db VXLAN_REMOTE_VTEP_VNI table";
+
+    list VXLAN_REMOTE_VTEP_VNI_LIST {
+        key "name mapname";
+
+        leaf name {
+            type leafref {
+                path "/svxlan:sonic-vxlan/svxlan:VXLAN_REMOTE_VTEP_LIST/svxlan:VXLAN_REMOTE_VTEP_VNI_LIST/svxlan:name";
+            }
+        }
+
+        leaf mapname {
+            type string;
+        }
+
+        leaf vni {
+            mandatory true;
+            type stypes:vnid_type;
+        }
+    }
+}
+
+
+#### Config DB Enhancements
+
+##### 2.1 VXLAN_REMOTE_VTEP_TABLE
+
+Producer: config manager
+
+Consumer: VxlanMgr
+
+Description: This table holds the tunnel name specifying the VTEP remote IP used for data plane configured tunnels.
+
+; New table
+
+key = VXLAN_REMOTE_VTEP_TABLE|remote_vtep_name
+                          ; VTEP name as a string
+; field = value
+remote_ip = ipv4_address ; Source Ipv4 address of the remote VTEP.
+
+
+##### 2.2 VXLAN_REMOTE_VTEP_VNI_TABLE
+
+Producer: config manager
+
+Consumer: VxlanMgr
+
+Description: This table specifies mapping of remote VTEP to VNI.
+
+Schema:
+
+; New table
+; specifies mapping of remote VTEP to VNI
+
+key = VXLAN_REMOTE_VTEP_VNI_TABLE|remote_vtep_name|tunnel_map_name
+                  ; Remote VTEP name + tunnel map name
+; field = value
+vni = 1*8DIGIT    ; vni that remote VTEP is member of.
+
+#### 3. APP_DB Enhancements
+
+##### 3.1 VXLAN_DATAPLANE_VTEP_TABLE
+
+Producer: config manager
+
+Consumer: VxlanMgr
+
+Description: This table holds the tunnel name specifying local VTEP's source IP used for Dataplane configured tunnels.
+
+Schema:
+
+;New table
+;holds the tunnel name specifying the VTEP source IP used for Dataplane configured tunnels
+
+key = VXLAN_DATAPLANE_VTEP:nvo_name
+                          ; nvo or VTEP name as a string
+; field = value
+source_vtep = tunnel_name
+
+
+##### 3.2 VXLAN_DATAPLANE_VTEP_VNI_TABLE
```

Comment:
The existing VXLAN_REMOTE_VNI table can be reused as is.
I am not clear on the need for this table as the VXLAN_REMOTE_VNI table and the OA handling of this 
table is sufficient.
VxlanMgr can populate this instead of fdbsyncd for static tunnel cases.

A "creation_source" field can be used to distinguish between VxlanMgr and fdbsyncd populated entries. 
This source can be used to infer if this tunnel is static or dynamic.

Reply:
I agree, it can be used. Currently, it is used by EvpnRemoteVnip2pOrch/EvpnRemoteVnip2mpOrch classes. Do you think these also can be reused (with naming change) or same table to be referenced again by Dataplane ?